### DISCUSSION TITLE
Set the background and font color of each cell while dynamically displaying columns

### DISCUSSION THREAD

**Discussion by wosiyourfather** [AvaloniaUI/Avalonia#12650] [Q&A]
I have a Winform program ported here, and this is how I implemented it in Winform
![4](https://github.com/AvaloniaUI/Avalonia/assets/53590491/d0c0c29a-2938-4013-a69b-5693e3c1c7d5)
But in Avalonia, I thought about defining an entity to handle
![444](https://github.com/AvaloniaUI/Avalonia/assets/53590491/915982f2-061a-4f9e-9755-00862d83f348)
Then change its template
![44](https://github.com/AvaloniaUI/Avalonia/assets/53590491/d00fc16d-961c-4944-9986-a45ed0e92377)
Finally, dynamically load the columns
![4444](https://github.com/AvaloniaUI/Avalonia/assets/53590491/931d0891-fe40-40c6-8d56-36480c6d3b8c)
But it still doesn't achieve the effect I want, making every cell color change according to the bound field。

This is my demo. Can someone help me take a look
[AvaloniaDatagridBindingToDatatable.zip](https://github.com/AvaloniaUI/Avalonia/files/12425894/AvaloniaDatagridBindingToDatatable.zip)

**Comment by bsampica** [comment]
I'm not quite getting exactly what you're attempting from your code, and I didn't download the zip, maybe put it up in a GitHub repo, or if it's a private project, reproduce just this part of it in a repo that we can look at?

It sounds like you're trying to conditionally set the style of the cell, based on the contents of the cell, is that correct, or am I interpreting incorrectly?

**Reply by wosiyourfather** [reply]
Yes, but I need to have a cells method that can set every cell, just like in Winform

**Reply by timunie** [reply]
Avalonis is not like WinForms. You have to learn new concepts, or stick with WinForms. Sorry...

**Reply by wosiyourfather** [reply]
I just want to find a way to achieve the effect I want to modify (my requirement is a dynamic column that needs to be updated to the background color and color of each cell)

**Reply by timunie** [reply]
@wosiyourfather you are messing way too much with your code. Can you create a sketch of what your goal is?
